Add ssh key configuration for instance edit WD-19727

edlerd opened the pull request:
## Done

- Add ssh key configuration for instance edit

Fixes WD-19727

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - create an instance and apply ssh keys
    - create a profile with ssh keys
    - ensure the profiles ssh keys get applied

Todo:
* profile edit and create saving
* show inherited ssh keys
* move to dedicated section in config form

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://lxd-ui-1133.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/4180)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

piperdeck commented:
This is looking really polished! I just have a couple notes:

---

Can we hold onto text that's been typed into this field when you switch to another source and then switch back?

https://github.com/user-attachments/assets/2c8a8b65-43d2-4289-b41a-52beaa7167a1

---

Can we please have the file picker filter for `.pub` files?

![CleanShot 2025-03-24 at 22 09 22@2x](https://github.com/user-attachments/assets/7fd39e0c-dd0b-4aeb-8024-35cdcc4896bd)

---

After I create an ssh key and confirm, when the modal disappears the key I've just created ends up below the fold, so I can't see confirmation of what I've just done without scrolling down. It also hides the new ssh key button, so if I have a few keys to create in a row, I need to scroll down in between each one to click the button.

Can we please have it so that the page scrolls up to reveal the bottom of the ssh keys table and button after confirming key creation in the modal?

https://github.com/user-attachments/assets/66cf0ec0-cdbb-4e03-8ace-1e0d31ac738b

---

This feels like a small thing, but because ssh keys have this long string of text in them, the text wrapper ends up breaking them on the space after `ssh-rsa`, creating this weird white space that makes it look like you've mistakenly pasted something with two lines. Could we add `word-break: break-all` to this textarea so that it breaks the text inside the hash rather than on the space before the hash?

| Current | With word-break: break-all |
| --- | --- |
| ![CleanShot 2025-03-24 at 22 32 39@2x](https://github.com/user-attachments/assets/94d4f4d8-f350-4a45-9d8c-aa5ca2a46a10) | ![CleanShot 2025-03-24 at 22 32 30@2x](https://github.com/user-attachments/assets/3b35e5ee-fe3a-4574-94e4-6e5ed6f0875e) |

---

On safari, when I upload a public key file, the attached file has this weird dotted rectangle next to the file name. Is this normal? Is it supposed to be a file icon?

![CleanShot 2025-03-24 at 22 40 22@2x](https://github.com/user-attachments/assets/4f7c19ca-065b-4f80-98ad-94fc525f4e53)

edlerd commented:
Thanks for the review @piperdeck 

Addressed all issues mentioned. Please have another pass at the updated PR for design review.

Except for the "filter .pub files only", that was already set. I see on Ubuntu, if you walk up in the directory structure, the file filter is reset. But that is an OS behavior we can't influence. On macOS, it grays out the other files, works fine. On Ubuntu, it also works fine, as long as you stay in the directory or only walk down directories.

Kxiru reviewed (approved):
Thanks for this, David!

**Screenshots**:
1) Instance creation:
![image](https://github.com/user-attachments/assets/08d619b6-438e-4679-bbde-542e44838ef0)
![image](https://github.com/user-attachments/assets/aee55d7d-0ed1-40c5-9202-07430d116848)
![image](https://github.com/user-attachments/assets/3ab1f7ef-cf10-4ac3-a3e2-d4cab5f43b50)


2) Profile creation:
![image](https://github.com/user-attachments/assets/4882c556-8a78-4c2d-92d9-b677ac7b870b)

3) New instance inheriting Profile SSH key:
![image](https://github.com/user-attachments/assets/5ef00248-ce92-401c-bb5a-0ebfaeaadff4)
![image](https://github.com/user-attachments/assets/7b08b244-8ecf-4d53-9157-f0ca5a73732e)

Perhaps in a new PR we can display SSH keys on the Instance and Profile overview pages and maybe even the side panels?

edlerd commented:
> Perhaps in a new PR we can display SSH keys on the Instance and Profile overview pages and maybe even the side panels?

Oh yes, we could. At least an indication if keys are attached and how many.
